From: Mike Hommey <glandium@debian.org>
Date: Sat, 21 Jun 2008 03:09:21 +0200
Subject: Add another preferences directory for applications:
 preferences/syspref

This can be used to symlink to /etc/ice*/pref.

---
 mozilla/toolkit/xre/nsXREDirProvider.cpp |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/mozilla/toolkit/xre/nsXREDirProvider.cpp b/mozilla/toolkit/xre/nsXREDirProvider.cpp
index 200de65..6e11878 100644
--- a/mozilla/toolkit/xre/nsXREDirProvider.cpp
+++ b/mozilla/toolkit/xre/nsXREDirProvider.cpp
@@ -603,6 +603,7 @@ nsXREDirProvider::LoadAppBundleDirs()
 }
 
 static const char *const kAppendPrefDir[] = { "defaults", "preferences", nsnull };
+static const char *const kAppendSysPrefDir[] = { "defaults", "syspref", nsnull };
 
 #ifdef DEBUG_bsmedberg
 static void
@@ -644,6 +645,7 @@ nsXREDirProvider::GetFilesInternal(const char* aProperty,
     LoadAppDirIntoArray(mXULAppDir, kAppendPrefDir, directories);
     LoadDirsIntoArray(mAppBundleDirectories,
                       kAppendPrefDir, directories);
+    LoadAppDirIntoArray(mXULAppDir, kAppendSysPrefDir, directories);
 
     rv = NS_NewArrayEnumerator(aResult, directories);
   }
